cupy/cudnn.py

```diff
@@ -99,13 +88,20 @@ def create_uninitialized_tensor_descriptor():
     return desc
 
 
+_nd_tensor_cache = {}
```

Comment:
Yes. Because this cache is only added new object.
And, the Python has GIL. Each primitive operation is thread safe.